nixos/tests/endlessh-go.nix

```diff
@@ -0,0 +1,25 @@
+import ./make-test-python.nix ({ pkgs, ... }:
+  {
+    name = "endlessh-go";
+    meta = with pkgs.lib.maintainers; {
+      maintainers = [ azahi ];
+    };
+
+    nodes.machine = { ... }: {
+      services.endlessh-go = {
+        enable = true;
+        port = 2222;
+        prometheus = {
+          enable = true;
+          port = 9229;
+        };
+      };
+    };
+
+    testScript = ''
+      machine.start()
+      machine.wait_for_unit("endlessh-go.service")
+      machine.wait_until_succeeds("ss -ntl | grep -q 2222")
+      machine.wait_until_succeeds("ss -ntl | grep -q 9229")
```

Comment:
that's a good start. we'd try to use [specialisations](https://search.nixos.org/options?channel=unstable&show=specialisation&from=0&size=50&sort=relevance&type=packages&query=specialisation) to put multiple test configs into here. one that just checks whether the sockets answer anything useful (eg curl on the prometheus endpoint and `nc | dd bs=1 count=1` on the ssh endpoint), and another that does the same but with privileged ports. just to have a few basic checks that the module looks to be correct :)